lib/data/repositories/v1/product_repository.dart

```diff
@@ -10,11 +11,17 @@ class ProductRepository {
 
   ProductRepository(this._api, this._logger);
 
-  Future<Either<ApiError, List<ProductDto>>> getProducts() async {
+  /// Get purchaseable ticket products.
+  Future<Either<ApiError, List<Product>>> getProducts() async {
     final response = await _api.apiV1ProductsGet();
 
     if (response.isSuccessful) {
-      return Right(response.body!);
+      return Right(
+        response.body!
+            .map((productDto) => Product.fromDTO(productDto))
+            .where((productDto) => productDto.amount > 1)
```

Comment:
I think it stems from the semantics related issue #172.
This will be addressed by exposing two public methods in the cubit like this:
```dart
  /// Get purchaseable ticket (coffee card) products.
  Future<Either<ApiError, List<Product>>> getTicketProducts() async {
    return _getProducts(filter: (product) => product.amount > 1);
  }

  /// Get purchaseable single drink products.
  Future<Either<ApiError, List<Product>>> getSingleDrinkProducts() async {
    return _getProducts(filter: (product) => product.amount == 1);
  }
```

Reply:
I realize this adds (what could be) a redundant api call. I'll look into other solutions

Reply:
I believe reverting commit [2588e92](https://github.com/AnalogIO/coffeecard_app/pull/197/commits/2588e922aa86a9952d1ac8c8df94e8ae298405d0), either fully or in part, should be able to solve the issue.

Reply:
@TTA777 What do you think of the current implementation? I added a cache for all products in the product repository.

Reply:
Not a fan, and I don't think it would work all the time either. The product repository is registered as a factory in the service locator, so it is not guaranteed to get the same instance every time it is requested. That is, you may or may not get the cached data.

Secondly, I don't like introducing a state to the repository layer, which it essentially is. That should only be the domain of the cubits/ blocs in my eyes. Feel free to argue otherwise